src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java

```diff
@@ -189,9 +189,12 @@ public final Response search(@Context final Request request, @Context final Http
             @DefaultValue(DEFAULT_START_INDEX_AS_STRING) @QueryParam("start_index") final Integer startIndex,
             @DefaultValue(DEFAULT_SEARCH_RESULT_LIMIT_AS_STRING) @QueryParam("limit") final Integer limit) {
 
+        // Impose 1000 character search string limit internally
+        String internalSearchString = searchString.substring(0, Math.min(searchString.length(), 1000));
```

Comment:
I think this should be a constant like `SEARCH_TEXT_CHAR_LIMIT` of 1000, and then if the `searchString` is longer than that we return a 400 Bad Request and not truncate it.
The front end can do something nicer like warn you, and have an even shorter limit.

Reply:
Ahh I thought the idea was to silently truncate - I'll alter this